File: benchmarks/suite/sieve/bench.py
def main():
    n = 10000000
    sieve = bytearray(b'\x01' * n)
    sieve[0] = 0
    sieve[1] = 0

    i = 2
    while i * i < n:
        if sieve[i]:
            j = i * i
            while j < n:
                sieve[j] = 0
                j += i
        i += 1

    total = sum(sieve)
    print(f"Primes up to {n}: {total}")

main()

File: benchmarks/suite/sieve/bench.rs
fn main() {
    let n: usize = 10_000_000;
    let mut sieve = vec![true; n];
    sieve[0] = false;
    sieve[1] = false;

    let mut i = 2;
    while i * i < n {
        if sieve[i] {
            let mut j = i * i;
            while j < n {
                sieve[j] = false;
                j += i;
            }
        }
        i += 1;
    }

    let total: usize = sieve.iter().filter(|&&x| x).count();
    println!("Primes up to {}: {}", n, total);
}
